### ISSUE TITLE
please update 0.12 to npm ~

### ISSUE THREAD

**Issue by allmonday** [angular-ui/bootstrap#3021]
npm's angular-bootstrap is still 0.11~~~~

**Comment by leon** [comment]
@pkozlowski-opensource any news on this?

**Comment by funkytaco** [comment]
+1 looking for something more dependable than a temporary module

**Comment by kentcdodds** [comment]
Would really like this on npm!

**Comment by leon** [comment]
I've published it at angular-bootstrap https://www.npmjs.com/package/angular-bootstrap

it's version 0.12.0 because 0.13.0-SNAPSHOT was already uploaded, but it's the same.

I've also invited the maintainers for angular-ui so that they can update the repo.

**Comment by kentcdodds** [comment]
Thanks for taking initiative @leon

**Comment by karianna** [comment]
I'm going to reopen this so it's tracked on this side.

**Comment by kentcdodds** [comment]
@leon, could the "main" be `dist/ui-bootstrap-tpls.js` instead of `dist/ui-bootstrap-tpls.min.js`? Any time I need to debug in ui-bootstrap, it just shows me minified code, and that's not helpful. Anyone using this via npm will definitely have a build process that could/should minify the code anyway.

**Comment by leon** [comment]
There are a couple of differences between the package I've deployed and the standard angular-boostrap

1. It doesn't contain the version number in the dist/ files since npm handle the versions, the grunt and gulp build's shouldn't have to take into account that the filename of the generated files change.

2. it also contains an updated package.json file

check the pull request https://github.com/angular-ui/bootstrap/pull/3201

**Comment by leon** [comment]
@kentcdodds What is the community standard for the "main" attribute?

**Comment by kentcdodds** [comment]
**TL;DR** - main should point to a single unminified file (best if it uses UMD) that exports the angular module name.

As I've been converting a project from bower to npm, I've noticed some point to a `src` file. This makes me a *little* uncomfortable because if they ever decide to move where the source files are located, they need to make sure to update `main` and if they miss that step, they break everyone. I much rather prefer them to point to a `dist` folder file.

But in most cases, the file that's pointed to is an unminified file. In many cases it's a [UMD module](https://github.com/umdjs/umd). In an angular context, this isn't really important. However, I think the angular-ui-router project has it right when their export is the name of the module. This is nice because then I can do this:

```js
angular.module('myModule', [
  require('angular-ui-router')
]);
```

And I don't even need to know the name of the module. I think that I would consider this a best practice and would hope that the entire community follows this project.

**Comment by kentcdodds** [comment]
Also, I should have mentioned, but just to be clear. That single file should be browser shippable on its own. So it should be a concat of all the files necessary as well as all the templates necessary (via $templateCache for example) for the module to function.

**Comment by leon** [comment]
@kentcdodds  I'll change my pull request to point to the unminified version.

Does removing the version from the filename some how interfer with the bower release?

**Comment by kentcdodds** [comment]
The version should never be included as part of the filename because it prevents auto-updates from happening. Meaning, if I depend on version 1.0.1 then in my index.html (for example) I'll say:

```html
<script src="my-dependency-1.0.1.js"></script>
```

Then if the maintainer of the project pushes a version 1.0.2, it breaks me or I don't get it, because bower would automatically update the package for me and unless the maintainer pushes all the previous versions as well, I'm going to break or not get it until I change to:

```html
<script src="my-dependency-1.0.2.js"></script>
```

So the version should never be a part of the file name (unless you update the "main" but I would say that's just silly and just don't publish the file with the version number).

**Comment by leon** [comment]
@kentcdodds In my case the reference is in my gulp.js file, but the same applies there.

**Comment by kentcdodds** [comment]
@leon, just published this. Maybe it'll be helpful: [How to distribute your AngularJS module](https://medium.com/@kentcdodds/how-to-distribute-your-angularjs-module-e04d4dd58ddc)

**Comment by Foxandxss** [comment]
Done. Will update README soon.
